proposals/0083-relax-entry-constraints.md

```diff
@@ -0,0 +1,62 @@
+---
+simd: '0083'
+title: Relax Entry Constraints
+authors:
+  - Andrew Fitzgerald (Solana Labs)
+category: Standard
+type: Core
+status: Draft
+created: 2023-11-02
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+Remove the constraint that transaction entries must not conflict with each
+other.
+
+## Motivation
+
+The current constraint that transaction entries must not conflict with each
+other is unnecessary.
+It is a constraint, that made the current labs-client implementation easier,
+but is not required for the protocol to function correctly.
+The removal of this constraint simplifies the protocol, and allows more
+flexibility in the implementation of both block-produciton and
+block-validation.
+
+## Alternatives Considered
+
+1. Do nothing
+    - This is the simplest option, as we could leave the protocol as is.
+    However, this leaves the protocol more complex than it needs to be.
+
+## New Terminology
+
+None
+
+## Detailed Design
+
+Currently, if a transaction entry within a block contains transactions that
+conflict, the entire block is marked invalid.
+The proposal is that this constraint is removed entirely, and entries are
+allowed to contain transactions that conflict with each other.
+
+## Impact
+
+- The protocol is simplified
+- Block-production is more flexible
+
```

Comment:
While I don't disagree with your comments that a consensus altering change could fit many of the sections on this document (such as `Impact` / `Security` / `Drawbacks`), it could get repetitive to mention it multiple times. I'm wondering if mentioning it once is sufficient; maybe here given the prompt:
https://github.com/solana-foundation/solana-improvement-documents/blob/ffb3cca62752b4370578b9bbd44726ca736dd760/XXXX-template.md#L64-L67

Reply:
I suppose I was under the assumption that **all** SIMDs are consensus breaking. Is that not the case?

I was under the asssumption that a consensus change and feature-gating was implicit in the proposal; I can be explicit about that.

Reply:
> I suppose I was under the assumption that all SIMDs are consensus breaking. Is that not the case?

> I was under the asssumption that a consensus change and feature-gating was implicit all proposals here; I can be explicit about that

I haven't been involved with enough SIMD's to know 😬

Reply:
All feature gates should be consensus changing, SIMDs do not need to be. A SIMD can propose application standards or networking changes.

Reply:
f76e7d3: Added comments about consensus breakage under security considerations.